### DISCUSSION TITLE
Mentions | Changing items

### DISCUSSION THREAD

**Discussion by fredriktre** [ueberdosis/tiptap#4121] [General Questions]
I tried using the code from https://tiptap.dev/api/nodes/mention, but I had one problem.... I couldn't change the items part of suggestion.

I couldn't find anything that could explain it either, so just hoping that some smart fellas can help me here 😓.

Basically, I have my users in a mongodb cluster, and would need to at least get the usernames from there so I can add them in suggestions, making it possible to actually mention proper users.

though, in the code example given, it's inputted statically, so I would have to make it respond to whatever data it's given. 

```
import { ReactRenderer } from '@tiptap/react'
import tippy from 'tippy.js'
import MentionList from './MentionList.jsx'

export default {
    items: ({ query }) => {
        return [
            'Lea Thompson',
            'Cyndi Lauper',
            'Tom Cruise',
            'Madonna',
            'Jerry Hall',
            ....
    ]
      .filter(item => item.toLowerCase().startsWith(query.toLowerCase()))
      .slice(0, 5)
  },
.....
```
There isn't really any way to "import" strings for that items part nativly, and when I tried to add it. The code freaked out saying it wasn't a function.

so here we are... Anyone got any ideas?

I'll update here if I find a solution, just deprioritizing this in the meantime.

**Comment by hrishikesh-k** [comment]
You need to build the array yourself. For example:

```js
items: props => {
  return fetchFromMongoHere().then(mongoData => {
    return MongoData.map(mongoUser => {
      return mongoUser.username
    })
  })
}
```

**Comment by andredewaard** [comment]
If you didn't find a good answer, what i did was make the suggestion object a function that accepts an array of users.
```ts
export type OptionProps = {
  value: string | number | undefined
  label: string
}

const suggestion = (users: OptionProps[]): MentionOptions['suggestion'] =>  ({
  items: ({ query }) => {
    return users.filter(item => item.label.toLowerCase().startsWith(query.toLowerCase()))
  },
  ...
})
```
And in my component use it like this
```ts
Mention.configure({
  HTMLAttributes: {
    class: 'mention',
  },
  suggestion: suggestion(props.users ?? []),
}),
```
